### PR TITLE
Feat/HNT-241 - Next gen modal

### PR DESCRIPTION
**Pull request by archit-mahto**
## Related issues

See 

https://telusdigital.atlassian.net/browse/HNT-6
https://telusdigital.atlassian.net/browse/HNT-241

## Description

Update modals to match new wireframes:
* https://telus.invisionapp.com/share/9M112UQANJ8V#/screens
* https://telus.invisionapp.com/share/HF115FIMENX5#/screens/452741972_H3___Medium_Left_Body_Text

## Checklist before submitting pull request

- Commits follow our [Developer Guide](https://github.com/telus/tds-community/blob/chore/template-hints/.github/CONTRIBUTING.md)
- [x] New code is unit tested
- [x] make sure visual and accessibility tests pass
- [x] make sure code builds

### CODE REVIEW

File: packages/Modal/Footer/Footer.jsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
+/* eslint-disable no-unused-vars */
```
[the remark this answers is not in the document]
Comment (archit-mahto) [reply packages/Modal/Footer/Footer.jsx:1]: I can do that

File: packages/Modal/Modal.jsx [hunk 2]
```diff
@@ -27,21 +31,31 @@ const FocusTrap = withFocusTrap('div')
  */
```
Comment (MarielleRobertsTelusDigital) [inline packages/Modal/Modal.jsx:31]: delete comment

File: packages/Modal/styles.jsx [hunk 3]
```diff
@@ -125,3 +125,65 @@ export const StyledBox = styled(Box)({
     marginTop: 0,
   }),
 })
+
+export const HeaderWrapper = styled.div({
+  // paddingTop: '1rem',
```
Comment (MarielleRobertsTelusDigital) [inline packages/Modal/styles.jsx:130]: delete comment.

File: packages/Modal/styles.jsx [hunk 4]
```diff
@@ -125,3 +125,65 @@ export const StyledBox = styled(Box)({
...
+    .css({
+      marginTop: 0,
+      position: 'relative',
+      marginLeft: 'auto',
+      marginRight: 'auto',
+    }),
+})
+
+export const FooterWrapper = styled.div({
+  // marginTop: '1rem',
```
Comment (MarielleRobertsTelusDigital) [inline packages/Modal/styles.jsx:170]: delete unused code.

**Review by MarielleRobertsTelusDigital — CHANGES_REQUESTED** [review]
Great job! Just delete unused/commented code!

File: packages/Modal/Modal.jsx (hunk 2 above)
Comment (archit-mahto) [reply packages/Modal/Modal.jsx:31]: I don't see the comment in my file

**Review by KetanAgarwal — APPROVED** [review]
Looks fine to me.

**Review by MarielleRobertsTelusDigital — APPROVED** [review]

**Review by KetanAgarwal — COMMENTED** [review]
Why are we changing the whole modal as the impact will be high with such change ? Instead of changing the current community modal I think we should create a new package for the new modal. And for the current community modal just give them the option to change or update the variant of buttons from a prop.

**Pull request merged**